src/main/java/org/primefaces/component/datepicker/DatePickerBase.java

```diff
@@ -386,4 +387,35 @@ public void setOnYearChange(String onYearChange) {
     public boolean hasTime() {
         return this.isShowTime() || this.isTimeOnly() || super.hasTime();
     }
-}
\ No newline at end of file
+
+    @Override
+    public String calculatePattern() {
```

Comment:
Please add some test-methods for this method. Think we should cover usecases like usedefinied dateFormat with time (-> removeTime) and the calculated TimeOnlyPatterns for 12 and 24 hour-format.